### PR TITLE
Expose logout fields in /v3-public/authproviders endpoint

### PR DESCRIPTION
**Pull request by andreas-kupries**
## Issue: 

https://github.com/rancher/dashboard/issues/12314
 
## Problem

The standard user does not have permission to fetch the resource management.cattle.io.authconfig  which is crucial to the UI since it contains all the information regarding the type of logout we should do based on the flags logoutAllEnabled, logoutAllForced, logoutAllSupported . 
 
## Solution

Expose the `logoutAll*` fields in `/v3-public/authproviders` endpoint.

This requires a security assessment.

## Testing

Given Okta auth providers is enabled
```sh
k get authconfig okta -o yaml | grep logout
logoutAllEnabled: false
logoutAllForced: false
logoutAllSupported: true
```

```sh
curl https://<rancher-url>/v3-public/authproviders/okta
{
  "actions": {
    "login": "https://<rancher-url>/v3-public/oktaProviders/okta?action=login"
  },
  "baseType": "authProvider",
  "creatorId": null,
  "id": "okta",
  "links": {
    "self": "https://<rancher-url>/v3-public/oktaProviders/okta"
  },
  "logoutAllEnabled": false,
  "logoutAllForced": false,
  "logoutAllSupported": true,
  "redirectUrl": "https://<rancher-url>/v1-saml/okta/login",
  "type": "oktaProvider"
}
```

### Automated Testing

* Test types added/modified:
    * Unit

### Regressions Considerations

N/A

Existing / newly added automated tests that provide evidence there are no regressions:
N/A

### CODE REVIEW

File: pkg/apis/management.cattle.io/v3/public_authn_types.go [hunk 1]
```diff
@@ -14,6 +14,20 @@ type AuthProvider struct {
...
+	// Flag. True when the auth provider supports a `Logout All` operation.
+	// Currently only the SAML providers do, with their `Single Log Out` flow.
+	LogoutAllSupported bool `json:"logoutAllSupported,omitempty"`
+
+	// Flag. True when the auth provider is configured to accept a `Logout All`
+	// operation. Can be set if and only if the provider supports `Logout All`
+	// (see AuthConfig.LogoutAllSupported).
+	LogoutAllEnabled bool `json:"logoutAllEnabled,omitempty"`
+
+	// Flag. Can be set if and only if `LogoutAllEnabled` (above) is set.
```
Comment (crobby) [inline pkg/apis/management.cattle.io/v3/public_authn_types.go:27]: "Can" be set, or "is only meaningful if LogoutAllEnabled is set to true"?

**Review by crobby — DISMISSED** [review]
lgtm, bonus points for the test addition

**Review by crobby — DISMISSED** [review]

**Review by crobby — APPROVED** [review]

**Review by JonCrowther — APPROVED** [review]

File: pkg/apis/management.cattle.io/v3/public_authn_types.go (hunk 1 above)
Comment (andreas-kupries) [reply pkg/apis/management.cattle.io/v3/public_authn_types.go]: Second, is only meaningful ...

**Comment by andreas-kupries** [comment]
## Validation Template

## Root Cause

A standard user does not have the permissions to get the AuthConfig resources containing the flags (logoutAll*) controlling SAML logout behaviour. This breaks the dashboard when used by a standard user, instead of an admin.

## What was fixed, or what change have occurred

AuthConfigs are already exposed in a very reduced form through public authprovider endpoints not requiring any permission. The code converting from auth config to auth provider structures was modified to expose the logoutAll* flags as well.

Note: This requires associated changes in the dashboard to use these endpoints for the logoutAll information.

## Areas or cases that should be tested

Login and logout for the dashboard.

### ...

## What areas could experience regressions

Login, logout

## Are the repro steps accurate/minimal?

**Pull request merged**